scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitUnbundleCommandTest.java

```diff
@@ -0,0 +1,102 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package sonia.scm.repository.spi;
+
+import com.google.common.io.ByteSource;
+import com.google.common.io.Files;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GitUnbundleCommandTest extends AbstractGitCommandTestBase {
+  private GitContext gitContext;
+  private GitUnbundleCommand unbundleCommand;
+
+  @BeforeEach
+  void initCommand() {
+    gitContext = mock(GitContext.class);
+    unbundleCommand = new GitUnbundleCommand(gitContext);
+  }
+
+  @Test
+  void shouldUnbundleRepositoryFiles(@TempDir Path temp) throws IOException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    TarArchiveOutputStream taos = new TarArchiveOutputStream(baos);
+    String filePath = "test-input";
+    String fileContent = "HeartOfGold";
+    addEntry(taos, filePath, fileContent);
+    taos.finish();
+
+    when(gitContext.getDirectory()).thenReturn(temp.toFile());
+    ByteSource byteSource = ByteSource.wrap(baos.toByteArray());
+    UnbundleCommandRequest unbundleCommandRequest = new UnbundleCommandRequest(byteSource);
+
+    unbundleCommand.unbundle(unbundleCommandRequest);
+
+    File createdFile = temp.resolve(filePath).toFile();
+    assertThat(createdFile).exists();
+    assertThat(Files.readLines(createdFile, StandardCharsets.UTF_8).get(0)).isEqualTo(fileContent);
+  }
+
+  @Test
+  void shouldUnbundleNestedRepositoryFiles(@TempDir Path temp) throws IOException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    TarArchiveOutputStream taos = new TarArchiveOutputStream(baos);
+    String filePath = "objects/pack/test-input";
+    String fileContent = "hitchhiker";
+    addEntry(taos, filePath, fileContent);
+    taos.finish();
+
+    when(gitContext.getDirectory()).thenReturn(temp.toFile());
+    ByteSource byteSource = ByteSource.wrap(baos.toByteArray());
+    UnbundleCommandRequest unbundleCommandRequest = new UnbundleCommandRequest(byteSource);
+
+    unbundleCommand.unbundle(unbundleCommandRequest);
+
+    File createdFile = temp.resolve(filePath).toFile();
+    assertThat(createdFile).exists();
+    assertThat(Files.readLines(createdFile, StandardCharsets.UTF_8).get(0)).isEqualTo(fileContent);
+  }
```

Comment:
This test method more or less looks like the first. Some abstractions would be nice to make the differences visible.